lib/process-services-cloud/src/lib/form/services/content-cloud-node-selector.service.spec.ts

```diff
@@ -0,0 +1,108 @@
+    let service: ContentCloudNodeSelectorService;
+    let apiService: AlfrescoApiService;
+    let notificationService: NotificationService;
+    let getNodeSpy: jasmine.Spy;
+    let dialog: MatDialog;
+    let openDialogSpy: jasmine.Spy;
+    let showWarningSpy: jasmine.Spy;
+
+    const relativePathNodeResponseBody = {
+        entry: {
+            id: 'mock-relative-path-node-id'
+        }
+    };
+
+    const aliasNodeResponseBody = {
+        entry: {
+            id: 'mock-alias-node-id'
+        }
+    };
+
+    setupTestBed({
+        imports: [TranslateModule.forRoot(), ProcessServiceCloudTestingModule, MatDialogModule]
+    });
+
+    beforeEach(() => {
+        service = TestBed.inject(ContentCloudNodeSelectorService);
+        notificationService = TestBed.inject(NotificationService);
+        apiService = TestBed.inject(AlfrescoApiService);
+        dialog = TestBed.inject(MatDialog);
+
+        showWarningSpy = spyOn(notificationService, 'showWarning');
+        openDialogSpy = spyOn(dialog, 'open').and.returnValue({
+            afterOpened: () => of({}),
+            afterClosed: () => of({}),
+            componentInstance: {
+                error: new Subject<any>()
+            }
+        });
+        getNodeSpy = spyOn(apiService.nodesApi, 'getNode');
+    });
+
+    it('should be able to open the content node select panel dialog', () => {
+        service.openUploadFileDialog('nodeId', 'single', true, true);
+        expect(openDialogSpy).toHaveBeenCalled();
+    });
+
+    it('should be able to show an notification if the relative path is invalid/deleted', async () => {
+        getNodeSpy.and.returnValue(Promise.reject('Not exists'));
+        const relativePathNodeEntry = await service.fetchNodeFromRelativePath('mock-alias', { relativePath: 'mock-relativePath' });
+        service.openUploadFileDialog('nodeId', 'single', true, true).subscribe(() => {
+            expect(openDialogSpy).toHaveBeenCalled();
+            expect(service.sourceNodeNotFound).toBe(true);
+            expect(showWarningSpy).toHaveBeenCalledWith('success');
+            expect(relativePathNodeEntry).toBeNull();
+        });
+    });
+
+    it('should not be able to show an notification if the relative path is valid', () => {
+        getNodeSpy.and.returnValue(Promise.resolve(relativePathNodeResponseBody));
+        service.openUploadFileDialog('nodeId', 'single', true, true);
+
+        expect(openDialogSpy).toHaveBeenCalled();
+        expect(service.sourceNodeNotFound).toBe(false);
+        expect(showWarningSpy).not.toHaveBeenCalled();
+    });
+
+    it('should be able to fetch given relative path node id', async () => {
+        getNodeSpy.and.returnValue(Promise.resolve(relativePathNodeResponseBody));
+        await service.fetchNodeFromRelativePath('mock-alias', { relativePath: 'mock-relativePath' });
+
+        expect(getNodeSpy).toHaveBeenCalledWith('mock-alias', {
+            relativePath: 'mock-relativePath'
+        });
```

Comment:
I have the feeling we can enrich those tests, instead of calling a method manually and expect for something to be returned or called can we test something more meaningful? Here we are checking that a wrapper method performs the API call with the right parameters. I am not fully into the context and I might be wrong but please take a moment to think if we can test something more meaningful.

Reply:
Actually, this test is not checking mocked responses. It's checking getNodeSpy(JS-API) has been called with the right parameters.